dataclients/kubernetes/definitions/routegroupvalidator.go

```diff
@@ -91,6 +93,19 @@ func (rgv *RouteGroupValidator) predicatesValidation(item *RouteGroupItem) error
 	return errorsJoin(errs...)
 }
 
+func (rgv *RouteGroupValidator) validateBackends(item *RouteGroupItem) error {
+	var errs []error
+	for _, backend := range item.Spec.Backends {
+		address, err := url.Parse(backend.Address)
+		if err != nil {
+			errs = append(errs, fmt.Errorf("failed to parse backend address %q: %w", backend.Address, err))
+		} else if address.Path != "" {
```

Comment:
Maybe also check for query?

Reply:
I was checking if we can use queryparam without at least one path so `example.com?foo=bar` vs `example.com/?foo=bar`. With url.Parse https://go.dev/play/p/iRU9__MGc8S you can achieve both so it makes sense to also detect this case.